module/VuFind/src/VuFind/View/Helper/Root/Icon.php

```diff
@@ -0,0 +1,219 @@
+     * HeadLink helper
+     *
+     * @var HeadLink
+     */
+    protected $headLink;
+
+    /**
+     * Prevent extra work by only appending the stylesheet once
+     *
+     * @var boolean
+     */
+    protected $styleAppended = false;
+
+    /**
+     * Constructor
+     *
+     * @param ThemeInfo      $themeInfo    Theme info helper
+     * @param CacheManager   $cacheManager Cache manager instance
+     * @param EscapeHtmlAttr $escAttr      EscapeHtmlAttr view helper
+     * @param HeadLink       $headLink     For stylesheet appending
+     */
+    public function __construct(
+        ThemeInfo $themeInfo, CacheManager $cacheManager,
+        EscapeHtmlAttr $escAttr, HeadLink $headLink
+    ) {
+        $this->config = $themeInfo->getMergedConfig('icons', true);
+        $this->defaultSet = $this->config['defaultSet'] ?? 'FontAwesome';
+        $this->iconMap = $this->config['aliases'] ?? [];
+        $this->cache = $cacheManager->getCache('object', 'iconHelper');
+        $this->esc = $escAttr;
+        $this->headLink = $headLink;
+    }
+
+    /**
+     * Map icon to set. Add prefix, return with set and template.
+     * Broken out for easier customization.
+     *
+     * @param string $name Icon name or key from theme.config.php
+     *
+     * @return array
+     */
+    protected function mapIcon(string $name): array
+    {
+        $icon = $this->iconMap[$name] ?? $name;
+        $set = $this->defaultSet;
+
+        // Override set from config (ie. FontAwesome:icon)
+        if (strpos($icon, ':') !== false) {
+            [$set, $icon] = explode(':', $icon, 2);
+        }
+
+        // Find set in theme.config.php
+        $setConfig = $this->config['sets'][$set] ?? [];
+        $template = $setConfig['template'] ?? $set;
+        $prefix = $setConfig['prefix'] ?? '';
+
+        return [$prefix . $icon, $set, $template];
+    }
+
+    /**
+     * Reduce extra parameters to one attribute string.
+     * Broken out for easier customization.
+     *
+     * @param array $extra Extra HTML attributes for the HTML tag
+     *
+     * @return string
+     */
+    protected function compileAttrs(array $extra): string
+    {
+        $attrs = '';
+        foreach ($extra as $key => $val) {
+            $attrs .= ' ' . $key . '="' . ($this->esc)($val) . '"';
+        }
+        return $attrs;
+    }
+
+    /**
+     * Create a unique key for icon names and extra attributes
+     *
+     * @param string $name  Icon name or key from theme.config.php
+     * @param array  $extra Extra HTML attributes for the HTML tag
+     *
+     * @return string
+     */
+    protected function cacheKey(string $name, $extra = []): string
+    {
+        if (empty($extra)) {
+            return $name;
+        }
+        ksort($extra);
+        return $name . '+' . md5(json_encode($extra));
+    }
+
+    /**
+     * Returns inline HTML for icon
+     *
+     * @param string $name  Which icon?
+     * @param array  $extra Just extra HTML attributes for now
```

Comment:
Should we make this comment more final and confident, or do you still think this might be used for something more in future?